internal/persistence/snapshot.go

```
@@ -0,0 +1,455 @@
+// snapshotForkDump triggers a fork-based snapshot. The parent serializes the
+// engine map to a pipe under a brief read lock, then ForkExec's the same binary
+// with --snapshot-child. The child reads from the pipe and writes the snapshot
+// file. If fork fails, falls back to an in-process write.
+func snapshotForkDump(dir string, shardID uint32, engine *storage.Engine, logger log.Logger) error {
+	pr, pw, err := os.Pipe()
+	if err != nil {
+		_, err = snapshotWrite(dir, shardID, engine, logger)
+		return err
+	}
+
+	cmd := exec.Command(os.Args[0], "--snapshot-child")
```

golangci-lint (2.12.2)

[error] os/exec.Command must not be called. use os/exec.CommandContext

(noctx)